backend/tests/test_contact.py

```diff
@@ -56,3 +56,32 @@ def test_delete_a_contact(self):
         # deleting a contact that does not exist
         response = self.client.delete("/contact/{}".format(c_id))
         self.assertEqual(response.status_code, 404)
+
+    # testing creating a contact
+    def test_create_a_contact(self):
+        dummy_ct_id = uuid.uuid4()
+        dummy_ct = ContactType(
+            id=dummy_ct_id,
+            hex_colour="#ffffff",
+            type="dummy type",
+            description="dummy description",
+        )
+
+        create_obj = {
+            "name": "dummy name",
+            "email": "dummy email",
+            "secondary_email": "dummy secondary email",
+            "cellphone": "dummy cell phone",
+            "role": "dummy role",
+            "organization": "dummy organization",
+            "neighbourhood": "dummy neighbourhood",
+            "contact_type": dummy_ct,
```

Comment:
@dgiltz The reason why your tests are failing, is because `dummy_ct` is of type `ContactType`. Which is not serializable when passed through the endpoint. A potential way of doing is this to provide the id of the `contact_type` that this contact belongs to, and thus in your endpoint, query for the `contact_type` the id is referring to and add the `Contact` to its `contacts` attribute, or the other way around of adding the queried `contact_type` to the newly created `Contact`'s `contact_types` attribute.

Reply:
The problem is though that because ContactType is a non-nullable field for creating a contact, I have it set so that you cannot create a contact unless that field is filled. So to test the create a contact route I have to have something in Contact Type or else I get a 400 status code. Does this mean that I should change how the route is implemented? Because without a valid contact type as input I cannot create a contact as currently implemented.

Reply:
@dgiltz That is correct. For your unit test, you can reference 
https://github.com/hack4impact-mcgill/mu-crm-tool/blob/main/backend/tests/test_contact.py#L22
on how to create a `Contact` and how to attach its `ContactType`. 
As for the route, yes you would probably have to make changes to it. If you take a look at the design (go to the figma board), you can see that there is a drop down button to choose a `ContactType` for the `Contact` you're creating. This allows us to attach a `ContactType`'s attribute (be it `type` or `id`) to the body of the POST request of creating a `Contact`. Upon receiving the attribute (`type` or `id`), we can then query for the specific `ContactType` it is referring to, and append the newly created `Contact` to its list of `contacts`. Feel free to dm me for a quick chat as maybe a visual explanation would be better.

Reply:
Ok so I should be editing the route so that it takes a contact type's id in that field as well as a contact